example/example_test.go

```diff
@@ -45,6 +45,7 @@ func helper2(log logr.Logger, msg string) {
 
 func ExampleNew() {
 	zerolog.SetGlobalLevel(zerolog.DebugLevel)
+	zerolog.LevelFieldName = ""
```

Comment:
You have to add this to get the test to pass, right? That implies that the change in `NewLogSink` is a breaking change.

We'll need a `NewLogSinkWithOptions` and leave `NewLogSink` unchanged.

Reply:
Normally I agree that we need prevent breaking change, but this PR is to fix a bug that users can see any level or v field in the codepath not using logr/zerologr. And restoring level field is less likely to break code than removing field.